### PR TITLE
AF-1874 Add string cut for library widgets and breadcrumb Link

### PR DESCRIPTION
**Pull request by ohonchar**
* Added limit for input characters  in gav fields in Project settings and in creation project. 
* Fixed html layout in library widgets.

### CODE REVIEW

**Comment by domhanak** [comment]
Nice! Could you please add tests? :)

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties [hunk 1]
```diff
@@ -486,3 +486,4 @@ SingleOwnerIsMandatory=There always has to be at least one owner.
 SpaceOwnerChanged=The space owner has changed.
 DuplicatedContributor=The selected user is already a contributor.
 ContributorTypeNotAllowed=You do not have permission to save a contributor of this type.
+FieldCharacterLimit=Field "{0}" could not exceed 80 characters.
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: Not sure what the full message ends up reading, but should the "could" be "should" here? Depends what it looks like in the UI.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/ValidationServiceImpl.java [hunk 2]
```diff
@@ -215,4 +217,9 @@ public boolean validateGAVVersion(final String version) {
     public Collection<ValidationMessage> validateForDelete(final Path path) {
         return (Collection<ValidationMessage>) deleteValidators.stream().filter(v -> v.accept(path)).flatMap(c -> c.validate(path).stream()).collect(Collectors.toList());
     }
+
+    @Override
+    public boolean validateLimitCharacters(final String character) {
+        return character.length() < limitFieldCharacters;
```
Comment (Rikkola) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/ValidationServiceImpl.java]: Test this

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java [hunk 3]
```diff
@@ -255,6 +266,12 @@ private void validateName(final String name,
             return;
         }
 
+        if (name.length() >= limitFieldCharacters) {
+            endProjectCreation();
+            view.showError(view.getFieldLimitNameMessage());
+            return;
+        }
+
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java]: Test this

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java [hunk 4]
```diff
@@ -275,6 +292,12 @@ private void validateGroupId(final String groupId,
             return;
         }
 
+        if (groupId.length() >= limitFieldCharacters) {
+            endProjectCreation();
+            view.showError(view.getFieldLimitGroupIdMessage());
+            return;
+        }
+
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java]: Test this

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java [hunk 5]
```diff
@@ -295,6 +318,12 @@ private void validateArtifactId(final String artifactId,
             return;
         }
 
+        if (artifactId.length() >= limitFieldCharacters) {
+            endProjectCreation();
+            view.showError(view.getFieldLimitArtifactIdMessage());
+            return;
+        }
+
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java]: Test this

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java [hunk 6]
```diff
@@ -315,6 +344,12 @@ private void validateVersion(final String version,
             return;
         }
 
+        if (version.length() >= limitFieldCharacters) {
+            endProjectCreation();
+            view.showError(view.getFieldLimitVersionMessage());
+            return;
+        }
+
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/AddProjectPopUpPresenter.java]: And test this

**Comment by ohonchar** [comment]
jenkins restart the build

**Comment by ohonchar** [comment]
jenkins restart the build

**Comment by manstis** [comment]
@ohonchar I think you need to do something like:

Jenkins (please) retest this.

**Comment by ohonchar** [comment]
> @ohonchar I think you need to do something like:
> 
> Jenkins (please) retest this.

thnx a lot, Next time will be polite. :)

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties (hunk 1 above)
Comment (domhanak) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: imho `should` is the right word in this case

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by ohonchar** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
@Rikkola Do you mind checking again?

**Comment by Rikkola** [comment]
I don't mind. Looks good.

**Comment by ohonchar** [comment]
@manstis could u please review this?

**Comment by ederign** [comment]
Hi @manstis this PR breaks our library! I just ping you about this on gchat. :(

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/LibraryView.less [hunk 7]
```diff
@@ -373,6 +373,10 @@
 
     .card-pf-view .card-pf-title {
         margin: 0;
+        text-overflow: ellipsis !important;
+        overflow: hidden;
+        white-space: nowrap;
+        width:350px;
```
Comment (ederign) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/LibraryView.less]: most of times, setting hard code widths are a bad practice.

Comment (srambach) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/LibraryView.less]: Putting a width here will nearly always break at some screen width and risks the content overflowing the container. But, if you can add `overflow:hidden;` to the plain div that is wrapping the title and description, then you should have flexible ellipsing.

**Pull request merged**